homeassistant/components/lock/sesame.py

```diff
@@ -25,46 +25,55 @@
 
 
 # pylint: disable=unused-argument
-def setup_platform(hass, config: ConfigType,
-                   add_devices: Callable[[list], None], discovery_info=None):
+def setup_platform(
+        hass, config: ConfigType,
+        add_devices: Callable[[list], None], discovery_info=None):
     """Set up the Sesame platform."""
     import pysesame
 
     email = config.get(CONF_EMAIL)
     password = config.get(CONF_PASSWORD)
 
-    add_devices([SesameDevice(sesame) for
-                 sesame in pysesame.get_sesames(email, password)])
+    add_devices([SesameDevice(sesame) for sesame in
+                 pysesame.get_sesames(email, password)],
+                update_before_add=True)
 
 
 class SesameDevice(LockDevice):
     """Representation of a Sesame device."""
 
     _sesame = None
 
+    # Cached sesame properties
+    _device_id = None
```

Comment:
Declare these inside of `__init__` instead of here. Variables declared under the class are static for all instances of the class. We can get rid of the `_sesame = None` above as well. Should just look like this in `__init__`:
```python
self._sesame = sesame
self._device_id = None
self._nickname = None
...
```